api/app/models/job_file.rb

```diff
@@ -0,0 +1,214 @@
+
+        # Skip directories and symbolic links
+        # NOTE: Following symbolic links may cause security issues if not handled with care
+        #       For the time being it's best to ignore them
+        next if perm.include?('d') || perm.include?('l')
+
+        # Generate the relative path and file_id
+        rel_path = Pathname.new(name).expand_path(current_dir).relative_path_from(results_dir).to_s
+        file_id = Base64.urlsafe_encode64(rel_path)
+        id = "#{job_id}.#{file_id}"
+
+        # Load or build and cache a new file
+        get_from_cache(id) || JobFile.new(job_id, file_id, user: user, size: size).tap do |file|
+          set_in_cache(id, file)
+        end
+      end.reject(&:nil?).sort
+    end
+
+    def find!(id, **opts)
+      job_id, file_id = id.split('.', 2)
+      new(job_id, file_id, user: opts[:user]).tap do |job_file|
+        return nil unless job_file.exists?
+      end
+    end
+  end
+
+  # NOTE: The file_id could be one of the following:
+  # * 'stdout' - Denotes the standard output,
+  # * 'stderr' - Denotes the standard error], or
+  # * 'encoded_string' - The relative path to the file as a urlsafe base64 encoded string
+  #
+  # PS: The strings 'stdout'/'stderr' are invalid inputs to urlsafe_decode64 and thus
+  # can be used as special case inputs.
+  def initialize(job_id, file_id, user:, size: nil)
+    @job_id = job_id
+    @file_id = file_id
+    @user = user
+    @size = size
+  end
+
+  def id
+    "#{@job_id}.#{@file_id}"
+  end
+
+  # NOTE: The size maybe cached from the ls command, which allows the payload
+  # to be lazy loaded. However the size should be recalculated if the payload
+  # is being sent with the request.
+  #
+  # This edge case *may* happen surprisingly often as it is possible to retrieve
+  # files of a RUNNING command.
+  #
+  # Doing the recalculation implicitly is not an option because it makes the
+  # serializer order dependent. Explicit recalculation prevents this.
+  def size(recalculate: false)
+    @size = nil if recalculate
+    @size ||= payload.length
+  end
+
+  def exists?
+    payload # Attempts to load the payload
+    true
+  rescue FlightJobScriptAPI::CommandError
+    # The file does not exists due to a bad name
+    return false unless payload_command
+
+    # The file or job legitimately does not exist
+    return false if [20, 23].include? payload_command.exitstatus
+
+    # Something else has gone wrong, re-raise the error
+    raise $!
+  end
+
+  def find_job
+    Job.find!(@job_id, user: @user)
+  end
+
+  def decoded_file_id
+    @decoded_file_id unless @decoded_file_id.nil?
+    return @decoded_file_id = false if ['stdout', 'stderr'].include?(@file_id)
+    @decoded_file_id ||= Base64.urlsafe_decode64(@file_id)
+  rescue ArgumentError
+    # urlsafe_decode64 may raise ArgumentError if @file_id is invalid
+    @decoded_file_id = false
+  end
+
+  def path
+    @path unless @path.nil?
```

Comment:
Is this line missing a `return`?

Reply:
https://github.com/openflighthpc/flight-job-script-service/pull/31/commits/85f024b23ca7f77b2759b6a9f84dd75f9d14a7ef